### PR TITLE
Cmdlets for Express Route Cross Connections

### PR DESCRIPTION
**Pull request by vinayada1**
## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by markcowl** [comment]
@vinayada1 A really large set of network tests are failing, please fix.

File: src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/RemoveAzureRMExpressRouteCrossConnectionPeeringCommand.cs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+using MNM = Microsoft.Azure.Management.Network.Models;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet(VerbsCommon.Remove, "AzureRMExpressRouteCrossConnectionPeering"), OutputType(typeof(PSExpressRouteCrossConnection))]
+    public class RemoveAzureRMExpressRouteCrossConnectionPeeringCommand : NetworkBaseCmdlet
+    {
+        public const string IPv4 = "IPv4";
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/RemoveAzureRMExpressRouteCrossConnectionPeeringCommand.cs]: We should look at moving these definitions to [NetworkBaseCmdlet](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Network/Commands.Network/Common/NetworkBaseCmdlet.cs)

I see that its defined/used at least in one more place

File: src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/AzureRMExpressRouteCrossConnectionPeeringBase.cs [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+    using Microsoft.Azure.Commands.Network.Models;
+    using System.Management.Automation;
+    using MNM = Microsoft.Azure.Management.Network.Models;
+    using System.Linq;
+
+    public class AzureRMExpressRouteCrossConnectionPeeringBase : NetworkBaseCmdlet
+    {
+        public const string IPv4 = "IPv4";
+        public const string IPv6 = "IPv6";
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/AzureRMExpressRouteCrossConnectionPeeringBase.cs]: We should look at moving these definitions to [NetworkBaseCmdlet](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Network/Commands.Network/Common/NetworkBaseCmdlet.cs)

I see that its defined/used at least in one more place

**Comment by praries880** [comment]
ping @vinayada1 Do you need anything from  our end to move forward with the PR?

**Comment by vinayada1** [comment]
Need to fix help and checkin 19.0 SDK recordings. Haven't gotten around to doing that. Will do that soon.

**Comment by MiYanni** [comment]
I did not find a design review for these cmdlets. Please submit one to our design review repo via the process outlined here: https://github.com/Azure/adx-documentation-pr/wiki/PowerShell-On-boarding-Process#design-review

**Comment by vinayada1** [comment]
https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/68

File: src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/AzureRMExpressRouteCrossConnectionPeeringBase.cs (hunk 2 above)
Comment (vinayada1) [reply src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/AzureRMExpressRouteCrossConnectionPeeringBase.cs]: Done

File: src/Common/Commands.Common.Network/Version2017_10_01/INetworkManagementClient.cs [hunk 3]
```diff
@@ -104,6 +104,11 @@ public partial interface INetworkManagementClient : System.IDisposable
         /// </summary>
         IExpressRouteServiceProvidersOperations ExpressRouteServiceProviders { get; }
 
+        /// <summary>
+        /// Gets the IExpressRouteCrossConnectionsOperations.
+        /// </summary>
+        //IExpressRouteCrossConnectionsOperations ExpressRouteCrossConnections { get; }
```
[the remark this answers is not in the document]
Comment (vinayada1) [reply src/Common/Commands.Common.Network/Version2017_10_01/INetworkManagementClient.cs]: Done

**Comment by MiYanni** [comment]
@vinayada1 You need to add `TestExpressRouteCrossConnectionApis.json` to the Commands.Network.Test.csproj. Make sure it is set to PreserveNewest.

File: src/Common/Commands.Common.Network/Version2017_10_01/Models/ExpressRouteCircuitPeeringType.cs [hunk 4]
```diff
@@ -12,9 +12,9 @@ namespace Microsoft.Azure.Management.Internal.Network.Version2017_10_01.Models
 {
 
     /// <summary>
-    /// Defines values for ExpressRouteCircuitPeeringType.
+    /// Defines values for ExpressRoutePeeringType.
     /// </summary>
-    public static class ExpressRouteCircuitPeeringType
+    public static class ExpressRoutePeeringType
```
Comment (MiYanni) [inline src/Common/Commands.Common.Network/Version2017_10_01/Models/ExpressRouteCircuitPeeringType.cs]: You can't rename this class. Please change the name back. If you are having issues with your new classes, either use fully qualified names or alias usings.

File: src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/ExpressRouteCrossConnectionTests.ps1 [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+      #Assert-IsNull $crossConnection
+
+      #Verify List all cmdlet
+      #$crossConnectionsAll = Get-AzureRMExpressRouteCrossConnection
+      #Assert-AreEqual 0 @($crossConnectionsAll).Count
+
+      #Verify List by resource group cmdlet
+      #$crossConnectionsByRg = Get-AzureRMExpressRouteCrossConnection -ResourceGroupName $crossConnectionResourceGroupName
+      #Assert-AreEqual 0 @($crossConnectionsByRg).Count
```
Comment (MiYanni) [inline src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/ExpressRouteCrossConnectionTests.ps1]: Please remove commented code.

File: src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/AzureRMExpressRouteCrossConnectionPeeringBase.cs [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+            HelpMessage = "PeerAddressType")]
+        [ValidateSet(
+           IPv4,
+           IPv6,
+           IgnoreCase = true)]
+        public string PeerAddressType { get; set; }
+
+        public void ConstructMicrosoftConfig(PSExpressRouteCrossConnectionPeering peering)
+        {
+            if (this.MicrosoftConfigAdvertisedPublicPrefixes != null && this.MicrosoftConfigAdvertisedPublicPrefixes.Any())
```
Comment (MiYanni) [inline src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/AzureRMExpressRouteCrossConnectionPeeringBase.cs]: `this.` does nothing in any of these statements (including all the ones below). It is not needed and can be safely removed.

File: src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/GetAzureRMExpressRouteCrossConnectionPeeringCommand.cs [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipeline = true,
+            HelpMessage = "The cross connection")]
+        public PSExpressRouteCrossConnection ExpressRouteCrossConnection { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+            if (!string.IsNullOrEmpty(this.Name))
```
Comment (MiYanni) [inline src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/GetAzureRMExpressRouteCrossConnectionPeeringCommand.cs]: `this.` does nothing in any of these statements (including all the ones below). It is not needed and can be safely removed.

File: src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/NewAzureRMExpressRouteCrossConnectionPeeringCommand.cs [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+            ConfirmAction(
+               Force.IsPresent,
+               string.Format(Properties.Resources.OverwritingResource, Name),
+               Properties.Resources.CreatingResourceMessage,
+               Name,
+               () =>
+               {
+                   var peering = new PSExpressRouteCrossConnectionPeering();
+
+                   peering.Name = this.Name;
```
Comment (MiYanni) [inline src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/NewAzureRMExpressRouteCrossConnectionPeeringCommand.cs]: `this.` does nothing in any of these statements (including all the ones below). It is not needed and can be safely removed.

File: src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/RemoveAzureRMExpressRouteCrossConnectionPeeringCommand.cs [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+            base.Execute();
+
+            ConfirmAction(
+               Force.IsPresent,
+               string.Format(Properties.Resources.OverwritingResource, Name),
+               Properties.Resources.CreatingResourceMessage,
+               Name,
+               () =>
+               {
+                   var peering = this.ExpressRouteCrossConnection.Peerings.SingleOrDefault(resource => string.Equals(resource.Name, this.Name, System.StringComparison.CurrentCultureIgnoreCase));
```
Comment (MiYanni) [inline src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/RemoveAzureRMExpressRouteCrossConnectionPeeringCommand.cs]: `this.` does nothing in any of these statements (including all the ones below). It is not needed and can be safely removed.

File: src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/SetAzureRMExpressRouteCrossConnectionPeeringCommand.cs [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+            base.Execute();
+
+            ConfirmAction(
+               Force.IsPresent,
+               string.Format(Properties.Resources.OverwritingResource, Name),
+               Properties.Resources.CreatingResourceMessage,
+               Name,
+               () =>
+               {
+                   var peering = this.ExpressRouteCrossConnection.Peerings.SingleOrDefault(resource => string.Equals(resource.Name, this.Name, StringComparison.CurrentCultureIgnoreCase));
```
Comment (MiYanni) [inline src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/SetAzureRMExpressRouteCrossConnectionPeeringCommand.cs]: `this.` does nothing in any of these statements (including all the ones below). It is not needed and can be safely removed.

File: src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeeringStats/GetAzureRMExpressRouteCrossConnectionArpTableCommand.cs [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The DevicePath, can be either Primary or Secondary")]
+        [ValidateNotNullOrEmpty]
+        public DevicePathEnum DevicePath { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+            var arpTables = this.ExpressRouteCrossConnectionClient.ListArpTable(ResourceGroupName, CrossConnectionName, PeeringType, DevicePath.ToString()).Value.Cast<object>().ToList();
```
Comment (MiYanni) [inline src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeeringStats/GetAzureRMExpressRouteCrossConnectionArpTableCommand.cs]: `this.` does nothing in any of these statements (including all the ones below). It is not needed and can be safely removed.

File: src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeeringStats/GetAzureRMExpressRouteCrossConnectionRouteTableCommand.cs [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The DevicePath, can be either Primary or Secondary")]
+        [ValidateNotNullOrEmpty]
+        public DevicePathEnum DevicePath { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+            var routeTables = this.ExpressRouteCrossConnectionClient.ListRoutesTable(ResourceGroupName, CrossConnectionName, PeeringType, DevicePath.ToString()).Value.Cast<object>().ToList();
```
Comment (MiYanni) [inline src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeeringStats/GetAzureRMExpressRouteCrossConnectionRouteTableCommand.cs]: `this.` does nothing in any of these statements (including all the ones below). It is not needed and can be safely removed.

File: src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeeringStats/GetAzureRMExpressRouteCrossConnectionRouteTableSummaryCommand.cs [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The DevicePath, can be either Primary or Secondary")]
+        [ValidateNotNullOrEmpty]
+        public DevicePathEnum DevicePath { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+            var routeTableSummaries = this.ExpressRouteCrossConnectionClient.ListRoutesTableSummary(ResourceGroupName, CrossConnectionName, PeeringType, DevicePath.ToString()).Value.Cast<object>().ToList();
```
Comment (MiYanni) [inline src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeeringStats/GetAzureRMExpressRouteCrossConnectionRouteTableSummaryCommand.cs]: `this.` does nothing in any of these statements (including all the ones below). It is not needed and can be safely removed.

File: src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/ExpressRouteCrossConnectionBaseCmdlet.cs [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+
+                throw;
+            }
+
+            return true;
+        }
+
+        public PSExpressRouteCrossConnection GetExpressRouteCrossConnection(string resourceGroupName, string name)
+        {
+            var crossConnection = this.ExpressRouteCrossConnectionClient.Get(resourceGroupName, name);
```
Comment (MiYanni) [inline src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/ExpressRouteCrossConnectionBaseCmdlet.cs]: `this.` does nothing in any of these statements (including all the ones below). It is not needed and can be safely removed.

File: src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/GetAzureRMExpressRouteCrossConnectionCommand.cs [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+            ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+            if (!string.IsNullOrEmpty(this.Name))
+            {
+                var crossConnection = this.GetExpressRouteCrossConnection(this.ResourceGroupName, this.Name);
```
Comment (MiYanni) [inline src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/GetAzureRMExpressRouteCrossConnectionCommand.cs]: `this.` does nothing in any of these statements (including all the ones below). It is not needed and can be safely removed.

File: src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/SetAzureRMExpressRouteCrossConnectionCommand.cs [hunk 16]
```diff
@@ -0,0 +1,69 @@
...
+            HelpMessage = "Do not ask for confirmation if you want to overrite a resource")]
+        public SwitchParameter Force { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+
+            ConfirmAction(
+               Force.IsPresent,
+               string.Format(Properties.Resources.OverwritingResource, this.ExpressRouteCrossConnection.Name),
```
Comment (MiYanni) [inline src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/SetAzureRMExpressRouteCrossConnectionCommand.cs]: `this.` does nothing in any of these statements (including all the ones below). It is not needed and can be safely removed.

File: src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/AddAzureRMExpressRouteCrossConnectionPeeringCommand.cs [hunk 17]
```diff
@@ -0,0 +1,91 @@
...
+            base.Execute();
+
+            ConfirmAction(
+               Force.IsPresent,
+               string.Format(Properties.Resources.OverwritingResource, Name),
+               Properties.Resources.CreatingResourceMessage,
+               Name,
+               () =>
+               {
+                   var peering = this.ExpressRouteCrossConnection.Peerings.SingleOrDefault(resource => string.Equals(resource.Name, this.Name, System.StringComparison.CurrentCultureIgnoreCase));
```
Comment (MiYanni) [inline src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/AddAzureRMExpressRouteCrossConnectionPeeringCommand.cs]: `this.` does nothing in any of these statements (including all the ones below). It is not needed and can be safely removed.

File: src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/AzureRMExpressRouteCrossConnectionPeeringBase.cs [hunk 18]
```diff
@@ -0,0 +1,137 @@
...
+            Mandatory = false,
+            HelpMessage = "The SharedKey")]
+        [ValidateNotNullOrEmpty]
+        public string SharedKey { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The MircosoftConfigAdvertisedPublicPrefixes")]
+        [ValidateNotNullOrEmpty]
+        public List<string> MicrosoftConfigAdvertisedPublicPrefixes { get; set; }
```
Comment (MiYanni) [inline src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/AzureRMExpressRouteCrossConnectionPeeringBase.cs]: This should be singular `MicrosoftConfigAdvertisedPublicPrefix` and use type `string[]`.

File: src/ResourceManager/Network/Commands.Network/help/Add-AzureRmExpressRouteCrossConnectionPeering.md [hunk 19]
```diff
@@ -0,0 +1,320 @@
...
+Add-AzureRmExpressRouteCrossConnectionPeering -Name <String> -ExpressRouteCrossConnection <PSExpressRouteCrossConnection>
+ -PeeringType <String> -PeerASN <UInt32> -PrimaryPeerAddressPrefix <String> -SecondaryPeerAddressPrefix <String>
+ -VlanId <Int32> [-SharedKey <String>]
+ [-MicrosoftConfigAdvertisedPublicPrefixes <System.Collections.Generic.List`1[System.String]>]
+ [-MicrosoftConfigCustomerAsn <Int32>] [-MicrosoftConfigRoutingRegistryName <String>]
+ [-PeerAddressType <String>] [-LegacyMode <Boolean>] [-DefaultProfile <IAzureContextContainer>]
+ [<CommonParameters>]
+```
+
+### MicrosoftPeeringConfigRoutFilterId
```
Comment (MiYanni) [inline src/ResourceManager/Network/Commands.Network/help/Add-AzureRmExpressRouteCrossConnectionPeering.md]: Where are these 2 other parameter sets coming from? They are the same as the default.

File: src/ResourceManager/Network/Commands.Network/help/New-AzureRmExpressRouteCrossConnectionPeering.md [hunk 20]
```diff
@@ -0,0 +1,310 @@
...
+```
+New-AzureRmExpressRouteCrossConnectionPeering -Name <String> -PeeringType <String> -PeerASN <UInt32>
+ -PrimaryPeerAddressPrefix <String> -SecondaryPeerAddressPrefix <String> -VlanId <Int32> [-SharedKey <String>]
+ [-MicrosoftConfigAdvertisedPublicPrefixes <System.Collections.Generic.List`1[System.String]>]
+ [-MicrosoftConfigCustomerAsn <Int32>] [-MicrosoftConfigRoutingRegistryName <String>]
+ [-PeerAddressType <String>] [-LegacyMode <Boolean>] [-DefaultProfile <IAzureContextContainer>]
+ [<CommonParameters>]
+```
+
+### MicrosoftPeeringConfigRoutFilterId
```
Comment (MiYanni) [inline src/ResourceManager/Network/Commands.Network/help/New-AzureRmExpressRouteCrossConnectionPeering.md]: Where are these 2 other parameter sets coming from? They are the same as the default.

File: src/ResourceManager/Network/Commands.Network/help/New-AzureRmExpressRouteCrossConnectionPeering.md [hunk 21]
```diff
@@ -0,0 +1,310 @@
...
+    ResourceGroupName='ExpressRouteResourceGroup'
+    Location='West US'
+    SkuTier='Standard'
+    SkuFamily='MeteredData'
+    ServiceProviderName='Equinix'
+    Peering=$PeerConfig
+    PeeringLocation='Silicon Valley'
+    BandwidthInMbps=200
+}
+New-AzureRmExpressRouteCrossConnectionPeering @parameters
```
Comment (MiYanni) [inline src/ResourceManager/Network/Commands.Network/help/New-AzureRmExpressRouteCrossConnectionPeering.md]: This example is still incorrect.

File: src/Common/Commands.Common.Network/Version2017_10_01/Models/ExpressRouteCircuitPeeringType.cs (hunk 4 above)
Comment (vinayada1) [reply src/Common/Commands.Common.Network/Version2017_10_01/Models/ExpressRouteCircuitPeeringType.cs]: This was renamed in swagger

File: src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/ExpressRouteCrossConnectionTests.ps1 (hunk 5 above)
Comment (vinayada1) [reply src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/ExpressRouteCrossConnectionTests.ps1]: Done

File: src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/AzureRMExpressRouteCrossConnectionPeeringBase.cs (hunk 6 above)
Comment (vinayada1) [reply src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/AzureRMExpressRouteCrossConnectionPeeringBase.cs]: Done

File: src/Common/Commands.Common.Network/Version2017_10_01/Models/ExpressRouteCircuitPeeringType.cs (hunk 4 above)
Comment (MiYanni) [reply src/Common/Commands.Common.Network/Version2017_10_01/Models/ExpressRouteCircuitPeeringType.cs]: If the swagger was changed, that would be a new version. You can't just change the `Version2017_10_01` models to match the current swagger. I'm assuming there were other changes beyond renaming this one class to the swagger from that point in time. This is in the common Network project, so it is used by other non-Network modules. Unfortunately, I'm not sure of the process for generating new common versions. @praries880 Any insight?

File: src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/GetAzureRMExpressRouteCrossConnectionPeeringCommand.cs (hunk 7 above)
Comment (vinayada1) [reply src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/GetAzureRMExpressRouteCrossConnectionPeeringCommand.cs]: Done

File: src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/NewAzureRMExpressRouteCrossConnectionPeeringCommand.cs (hunk 8 above)
Comment (vinayada1) [reply src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/NewAzureRMExpressRouteCrossConnectionPeeringCommand.cs]: Done

File: src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/SetAzureRMExpressRouteCrossConnectionCommand.cs (hunk 16 above)
Comment (vinayada1) [reply src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/SetAzureRMExpressRouteCrossConnectionCommand.cs]: Done

File: src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/AzureRMExpressRouteCrossConnectionPeeringBase.cs (hunk 18 above)
Comment (vinayada1) [reply src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/AzureRMExpressRouteCrossConnectionPeeringBase.cs]: Need to keep this consistent with the BGP peering cmdlets under circuit

File: src/ResourceManager/Network/Commands.Network/help/Add-AzureRmExpressRouteCrossConnectionPeering.md (hunk 19 above)
Comment (vinayada1) [reply src/ResourceManager/Network/Commands.Network/help/Add-AzureRmExpressRouteCrossConnectionPeering.md]: Fixed

File: src/ResourceManager/Network/Commands.Network/help/New-AzureRmExpressRouteCrossConnectionPeering.md (hunk 20 above)
Comment (vinayada1) [reply src/ResourceManager/Network/Commands.Network/help/New-AzureRmExpressRouteCrossConnectionPeering.md]: Done

File: src/ResourceManager/Network/Commands.Network/help/New-AzureRmExpressRouteCrossConnectionPeering.md (hunk 21 above)
Comment (vinayada1) [reply src/ResourceManager/Network/Commands.Network/help/New-AzureRmExpressRouteCrossConnectionPeering.md]: Fixed

**Comment by vinayada1** [comment]
Fixed the review comments. Could you take another look?

File: src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/AzureRMExpressRouteCrossConnectionPeeringBase.cs (hunk 18 above)
Comment (MiYanni) [reply src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/AzureRMExpressRouteCrossConnectionPeeringBase.cs]: We don't allow grandfathered logic from prior cmdlets to break PowerShell design guidelines unless there is business justification to do so. The BGP cmdlets were probably made prior to us enforcing PowerShell design practices. What is recommended is adhering to the design guidelines for new cmdlets and then updating your old cmdlets in the next breaking change release.

Comment (vinayada1) [reply src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/AzureRMExpressRouteCrossConnectionPeeringBase.cs]: Fixed

File: src/Common/Commands.Common.Network/Version2017_10_01/Models/ExpressRouteCircuitPeeringType.cs (hunk 4 above)
Comment (vinayada1) [reply src/Common/Commands.Common.Network/Version2017_10_01/Models/ExpressRouteCircuitPeeringType.cs]: Sorry....did not realize this is 2017_10_01. Fixed

**Comment by cormacpayne** [comment]
@vinayada1 feel free to assign this back to @MiYanni whenever you are able to address the comments from the design review

**Comment by vinayada1** [comment]
Sorry....was OOF for a while. Have addressed the comments in the cmdlet design review.

File: src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeeringStats/GetAzureRMExpressRouteCrossConnectionArpTableCommand.cs [hunk 22]
```diff
@@ -0,0 +1,102 @@
...
+            "Primary",
+            "Secondary",
+            IgnoreCase = true)]
+        public DevicePathEnum DevicePath { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+            List<object> arpTables = null;
+            if (!string.IsNullOrWhiteSpace(CrossConnectionName))
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeeringStats/GetAzureRMExpressRouteCrossConnectionArpTableCommand.cs]: you don't really need this check... the param validation will make sure that this had a value that is not empty.

This comment applies to the other cmdlets as well

Comment (vinayada1) [reply src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeeringStats/GetAzureRMExpressRouteCrossConnectionArpTableCommand.cs]: Need to check whether the arguments have been specified by a cross connection reference or the Resource group name and resource name

**Comment by praries880** [comment]
on demand : https://azuresdkci.westus2.cloudapp.azure.com/job/powershell-demand/948/

**Pull request merged**